### PR TITLE
boards: arm: Seeeduino XIAO enable i2c interface on sercom2

### PR DESCRIPTION
**Pull request by leonardp**
This enables the i2c interface on the Seeeduino XIAO.

This is my first pull-request to this repository. I hope it's ok to just send a pull request without opening an issue beforehand.

Signed-off-by: Leonard Pollak <leonardp@tr-host.de>

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Please add a body to your commit messages, according to the [Commit Guidelines](https://docs.zephyrproject.org/latest/contribute/index.html#commit-guidelines)

**Comment by leonardp** [comment]
Thanks for the quick review. I've also added a "Signed Off-by" line to the message body.
I hope everything is in order now.

**Review by MaureenHelm — APPROVED** [review]

**Comment by Tubbles** [comment]
I recently looked into getting this upstreamed and found your PR! Good stuff :) However I noticed you are missing a section in the pinmux, will add an edit suggestion. Also: I've tested this code with the [expansion board oled](https://wiki.seeedstudio.com/Seeeduino-XIAO-Expansion-Board/#oled-display), which seems to work after a work around regarding #36857.

File: boards/arm/seeeduino_xiao/pinmux.c [hunk 1]
```diff
@@ -40,6 +40,25 @@ static int board_pinmux_init(const struct device *dev)
 #warning Pin mapping may not be configured
 #endif
 
+#if ATMEL_SAM0_DT_SERCOM_CHECK(2, atmel_sam0_i2c) && defined(CONFIG_I2C_SAM0)
+	/* SERCOM2 on SDA=PA08, SCL=PA09 */
+	pinmux_pin_set(muxa, 8, PINMUX_FUNC_D);
+	pinmux_pin_set(muxa, 9, PINMUX_FUNC_D);
+#endif
+
```
Comment (Tubbles) [inline boards/arm/seeeduino_xiao/pinmux.c:48]:
```c
#if ATMEL_SAM0_DT_SERCOM_CHECK(0, atmel_sam0_i2c) && defined(CONFIG_I2C_SAM0)
#warning Pin mapping may not be configured
#endif
```

**Review by Tubbles — CHANGES_REQUESTED** [review]
Slight change to the pin mux

**Review by Tubbles — CHANGES_REQUESTED** [review]
Should probably also include in `boards/arm/seeeduino_xiao_defconfig`:
```Kconfig
CONFIG_I2C=y
```
Edit: Or maybe not? SPI does not seem to be there either, so maybe let the application do it instead?

File: boards/arm/seeeduino_xiao/pinmux.c (hunk 1 above)
Comment (leonardp) [reply boards/arm/seeeduino_xiao/pinmux.c:48]: good catch. i updated the commit to include the missing lines

**Comment by leonardp** [comment]
> Should probably also include in `boards/arm/seeeduino_xiao_defconfig`:
> 
> ```
> CONFIG_I2C=y
> ```
> 
> Edit: Or maybe not? SPI does not seem to be there either, so maybe let the application do it instead?

I don't think it is needed but i might be wrong on this one.

**Comment by MaureenHelm** [comment]
> > Should probably also include in `boards/arm/seeeduino_xiao_defconfig`:
> > ```
> > CONFIG_I2C=y
> > ```  
> > Edit: Or maybe not? SPI does not seem to be there either, so maybe let the application do it instead?
> 
> I don't think it is needed but i might be wrong on this one.

That would have the effect of enabling I2C in all applications, even `samples/hello_world`, which doesn't need it. We usually leave it up to the application to set `CONFIG_I2C=y`

**Comment by Tubbles** [comment]
> We usually leave it up to the application to set `CONFIG_I2C=y`

Isn't it also supposed to be like that with the DT? Right now both the SPI and I2C are `status = "okay"`, which seemed odd to me

**Review by cfriedt — APPROVED** [review]

**Pull request merged**